### PR TITLE
GCP VM Instance Stop charts

### PR DESCRIPTION
**Pull request by neelanjan00**
Added charts and icons for the GCP `vm-instance-stop` experiment.

### CODE REVIEW

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    app.kubernetes.io/version: latest
+spec:
+  definition:
+    scope: Cluster
+    permissions:
+      - apiGroups: 
+          - "" 
+          - "batch" 
+          - "litmuschaos.io"
+        resources:
```
Comment (ispeakc0de) [inline charts/gcp/gcp-vm-instance-stop/experiment.yaml:21]: can we sync these permissions with rbac.yaml?

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+          - "chaosresults"
+        verbs: 
+          - "create" 
+          - "list" 
+          - "get" 
+          - "patch" 
+          - "update" 
+          - "delete" 
+          - "deletecollection"
+    image: ""
```
Comment (ispeakc0de) [inline charts/gcp/gcp-vm-instance-stop/experiment.yaml]:
```suggestion
    image: litmuschaos/go-runner:latest
```

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      value: 'parallel'
+
+    # provide the LIB
+    # only litmus supported
+    - name: LIB
+      value: 'litmus'
+
+    # period to wait before and after injection of chaos in sec
+    - name: RAMP_TIME
+      value: '0'
```
Comment (ispeakc0de) [inline charts/gcp/gcp-vm-instance-stop/experiment.yaml:66]:
```suggestion
      value: ''
```

File: charts/gcp/gcp-vm-instance-stop/gcp-vm-instance-stop.chartserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    categories: gcp
+spec:
+  displayName: gcp-vm-instance-stop 
+  categoryDescription: >
+    stops GCP VM instances before restarting them, post the chaos duration 
+  keywords: 
+    - "vm-instances" 
+    - "kubernetes" 
+    - "gcp" 
+    - "exec"
```
Comment (ispeakc0de) [inline charts/gcp/gcp-vm-instance-stop/gcp-vm-instance-stop.chartserviceversion.yaml]: we can remove it

File: charts/gcp/gcp.chartserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  name: gcp 
+  version: 0.1.0
+  annotations:
+    categories: gcp
+spec:
+  displayName: gcp chaos
+  categoryDescription: >
+    causes loss of a non-boot storage disk from a node or VM instance for a specified duration of time 
+  experiments:
+  - vm-disk-loss
```
Comment (uditgaurav) [inline charts/gcp/gcp.chartserviceversion.yaml]: disk loss?

Comment (neelanjan00) [reply charts/gcp/gcp.chartserviceversion.yaml]: Fixed

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml (hunk 2 above)
Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/experiment.yaml]: Tagged with `ci`; fixed

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml (hunk 3 above)
Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/experiment.yaml:66]: Fixed

File: charts/gcp/gcp-vm-instance-stop/gcp-vm-instance-stop.chartserviceversion.yaml (hunk 4 above)
Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/gcp-vm-instance-stop.chartserviceversion.yaml]: Removed exec

**Review by ispeakc0de — DISMISSED** [review]

**Review by ispeakc0de — DISMISSED** [review]

File: charts/gcp/gcp-vm-instance-stop/engine.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
+apiVersion: litmuschaos.io/v1alpha1
+kind: ChaosEngine
+metadata:
+  name: gcp-vm-chaos
+spec:
+  engineState: 'active'
+  chaosServiceAccount: gcp-vm-instance-stop-sa
+  jobCleanUpPolicy: 'retain'
```
Comment (uditgaurav) [inline charts/gcp/gcp-vm-instance-stop/engine.yaml]: can we remove the job cleanup policy? This is to keep it uniform.

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml [hunk 7]
```diff
@@ -0,0 +1,94 @@
+apiVersion: litmuschaos.io/v1alpha1
+description:
+  message: |
+    stops GCP VM instances and GKE nodes for a specified duration of time and later restarts them
```
Comment (uditgaurav) [inline charts/gcp/gcp-vm-instance-stop/experiment.yaml]:
```suggestion
    Stops GCP VM instances and GKE nodes for a specified duration of time and later restarts them
```

File: charts/gcp/gcp-vm-instance-stop/rbac.yaml [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+  verbs: ["create","list","get"]
+- apiGroups: ["batch"]
+  resources: ["jobs"]
+  verbs: ["create","list","get","delete","deletecollection"]
+- apiGroups: ["litmuschaos.io"]
+  resources: ["chaosengines","chaosexperiments","chaosresults"]
+  verbs: ["create","list","get","patch","update"]
+- apiGroups: [""]
+  resources: ["nodes"]
+  verbs: ["patch","get","list"]
```
Comment (uditgaurav) [inline charts/gcp/gcp-vm-instance-stop/rbac.yaml]: Do we need patch?

File: charts/gcp/gcp.chartserviceversion.yaml [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+kind: ChartServiceVersion
+metadata:
+  name: gcp 
+  version: 0.1.0
+  annotations:
+    categories: gcp
+spec:
+  displayName: gcp chaos
+  categoryDescription: >
+    gcp contains chaos to disrupt state of gcp resources running part of the gcp services
```
Comment (uditgaurav) [inline charts/gcp/gcp.chartserviceversion.yaml]:
```suggestion
    GCP  contains chaos to disrupt the state of GCP resources running part of the GCP services
```

File: charts/gcp/gcp-vm-instance-stop/engine.yaml (hunk 6 above)
Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/engine.yaml]: Ofcourse, fixed now

File: charts/gcp/gcp-vm-instance-stop/experiment.yaml (hunk 7 above)
Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/experiment.yaml]: updated

File: charts/gcp/gcp.chartserviceversion.yaml (hunk 9 above)
Comment (neelanjan00) [reply charts/gcp/gcp.chartserviceversion.yaml]: fixed

File: charts/gcp/gcp-vm-instance-stop/rbac.yaml (hunk 8 above)
Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/rbac.yaml]: I took reference from the `ec2-instance-delete-by-id` experiment, it had patch included hence I included it

Comment (neelanjan00) [reply charts/gcp/gcp-vm-instance-stop/rbac.yaml]: Shall I remove it?

**Review by uditgaurav — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**